src/NuGet.Services.Metadata.Catalog.Monitoring/ValidationCollector.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Result;
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test;
+using NuGet.Services.Metadata.Catalog.Persistence;
+using Microsoft.Extensions.Logging;
+using NuGet.Services.Metadata.Catalog.Helpers;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    /// <summary>
+    /// Runs a <see cref="PackageValidator"/> against collected packages.
+    /// </summary>
+    public class ValidationCollector : SortingIdVersionCollector
+    {
+        public ValidationCollector(
+            Storage auditingStorage, 
+            PackageValidator packageValidator, 
+            Uri index, 
+            Func<HttpMessageHandler> handlerFunc = null, 
+            ILogger<ValidationCollector> logger = null) 
```

Comment:
>logger [](start = 41, length = 6)

When will this be null? Seems like we'd have have an `ArgumentNullException` in that case when calling `LogInformation`.
[https://github.com/aspnet/Logging/blob/c149d48a52403759ef1e1ae932c5e0faf9e340d6/src/Microsoft.Extensions.Logging.Abstractions/LoggerExtensions.cs#L206](https://github.com/aspnet/Logging/blob/c149d48a52403759ef1e1ae932c5e0faf9e340d6/src/Microsoft.Extensions.Logging.Abstractions/LoggerExtensions.cs#L206)